modules/Channel.md

```diff
@@ -0,0 +1,125 @@
+## Channel Module
+
+A channel is a FIFO data structure used to communicate between different Chapel
+tasks. You can learn more about how go-channels work by following this
+[link](https://gobyexample.com/channels).
+Since the operations on the channel are blocking by default, it enables the user
+to send values from one task to another in a synchronized manner.
+Here is a simple example in Go.
+
+```
+func main() {
+	// make a channel for string values
+	messages := make(chan string)
+
+	// new task to send a value to the channel
+	// it will wait until other task receives
+	go func() { messages <- "ping" }()
+
+	// receive the sent value
+	msg := <-messages
+	fmt.Println(msg)
+}
+```
+
+The module will have a generic `Channel` class (or record) to store the channel
+buffer and support operations on them.
+
+* The main functions that the module should have
+
+	* `init(type eltType, size = 0)` : Specify the buffer data type and buffer size
+	for creating a Channel.
+
+	* `send(value : eltType)` : This function waits for a space in the buffer
+	and	sends the `value` to the channel.
+
+	* `recv(out value : eltType) : bool` : This function waits until there is
+	something in the buffer and stores the received entity in `value`. It also
+	returns the status (`false` if the channel is closed and its buffer is empty).
```

Comment:
Since this function returns `false` if the channel is closed and the buffer is empty, we need to be able to default-initialize the returned type. But, some types, such as `owned C` for a class C, do not support default initialization.

We can support channels that store values that cannot be default initialized if we make this `throw` instead of returning `false`. At that rate, you could make this function just be `proc recv(): eltType throws`.

@Yudhishthira1406 - I would like to know if the returning from `recv` is important in cases you have seen/written and if a try/catch for these cases would be awkward. I think we should consider supporting a no-parens `proc recv()` in addition to or instead of the `recv(out value): bool` version.